pkgs/development/libraries/opencl-headers/default.nix

```diff
@@ -1,21 +1,24 @@
-{ lib, stdenv, fetchFromGitHub
+{ lib, stdenv, fetchFromGitHub, cmake
 }:
 
 stdenv.mkDerivation rec {
   pname = "opencl-headers";
-  version = "2021.06.30";
+  version = "2022.05.18";
 
   src = fetchFromGitHub {
     owner = "KhronosGroup";
     repo = "OpenCL-Headers";
     rev = "v${version}";
-    sha256 = "sha256-MdKC48f1zhVAcHrqzrgT9iaYrHXurV8vDt+GnDroO9s=";
+    sha256 = "sha256-A2MKU1ppfCKroN4qKGKN9XJvkcGZJy+Kn8FvxceQv7o=";
   };
 
-  installPhase = ''
-    mkdir -p $out/include/CL
-    cp CL/* $out/include/CL
-  '';
+  nativeBuildInputs = [ cmake ];
+
+  patches = [
+    ./fix-cmakelists-darwin.patch
```

Comment:
Can you replace the patch file with

```
# part of https://github.com/KhronosGroup/OpenCL-Headers/pull/203
(fetchpatch { 
  url = "https://github.com/KhronosGroup/OpenCL-Headers/commit/7bdd9509899ef9558a71a795b5daa6f6811a165d.patch"; 
  sha256 = "<TODO>"; 
})
```

Reply:
The patch does three things:

- Fixes the `ln` flags
- Fixes the headers installation path for those Darwin-only commands (from `${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR}` to `${CMAKE_INSTALL_FULL_INCLUDEDIR}`
- And I'm sure they are linking `CL` from the source directory into `${CMAKE_INSTALL_FULL_INCLUDEDIR}/OpenCL` so I patched it too.

So even if I use your patch, I still need the custom patch for the other two points.

Reply:
So, the point is to not merge this again to re-trigger a big rebuild, better wait until the patches are upstreamed, right?

Ok, I'll create the PRs to upstream and then update this. But I can do it tomorrow, now it's too late in my TZ.

Reply:
Or I can put a `lib.optionals stdenv.isDarwin`, then when updating with the upstream patches it won't trigger a Linux mass rebuild.

Reply:
I updated the commit with your patch and a second patch (on top of your patch) fixing the install paths. I've also added the `lib.optionals stdenv.isDarwin` to the patches.

```
$  tree result
result
├── include
│   ├── CL
│   │   ├── cl.h
│   │   ├── cl_d3d10.h
│   │   ├── cl_d3d11.h
│   │   ├── cl_dx9_media_sharing.h
│   │   ├── cl_dx9_media_sharing_intel.h
│   │   ├── cl_egl.h
│   │   ├── cl_ext.h
│   │   ├── cl_ext_intel.h
│   │   ├── cl_gl.h
│   │   ├── cl_gl_ext.h
│   │   ├── cl_half.h
│   │   ├── cl_icd.h
│   │   ├── cl_layer.h
│   │   ├── cl_platform.h
│   │   ├── cl_va_api_media_sharing_intel.h
│   │   ├── cl_version.h
│   │   └── opencl.h
│   └── OpenCL -> CL
└── share
    └── cmake
        └── OpenCLHeaders
            ├── OpenCLHeadersConfig.cmake
            ├── OpenCLHeadersConfigVersion.cmake
            └── OpenCLHeadersTargets.cmake
```

(I built it on `master`, though, since staging makes me rebuild a ton of dependencies).

Now it's really too late.